lib/rules/no-irregular-whitespace/index.mjs

```diff
@@ -39,16 +50,19 @@ const IRREGULAR_WHITESPACES = [
 	'\u3000', // Ideographic Space
 ];
 
-const IRREGULAR_WHITESPACES_PATTERN = new RegExp(`([${IRREGULAR_WHITESPACES.join('')}])`);
+const IRREGULAR_WHITESPACES_PATTERN = new RegExp(`([${IRREGULAR_WHITESPACES.join('')}]+)`, 'g');
 
 /**
  * @param {string} str
- * @returns {string | null}
+ * @returns {Array<{index: number, length: number}>}
  */
 const findIrregularWhitespace = (str) => {
-	const result = IRREGULAR_WHITESPACES_PATTERN.exec(str);
-
-	return (result && result[1]) || null;
+	return Array.from(str.matchAll(IRREGULAR_WHITESPACES_PATTERN)).map((match) => {
+		return {
+			index: match.index,
+			length: match[0].length,
+		};
+	});
 };
```

Comment:
The old implementation only reported a single irregular whitespace per AST field.
Instead of reporting each contiguous instance.